xapian-applications/omega/docs/newformat.rst

```diff
@@ -0,0 +1,174 @@
+   ::
+
+     } else if (mimetype == "text/x-foo") {
+
+   The filename of the file is in ``file``. The code you add should at least extract the "body" text of the document into the C++ variable ``dump``. Optionally, you can also set ``title`` (the document's title), ``keywords`` (additional text to index, but not to show the user), ``sample`` (if set, this is used to generate the static document "snippet" which is stored; if not set, this is generated from dump) and ``topic``
+   ::
+
+     string tmpfile = get_tmpfile("tmp.html");
+     if (tmpfile.empty())
+       return;
+     string safetmp = shell_protect(tmpfile);
+     string cmd = "foo2utf16 --content " + shell_protect(file) + " " + safetmp;
+     try {
+       (void)stdout_to_string(cmd);
+       dump = file_to_string(tmpfile);
+       convert_to_utf8(dump, "UTF-16");
+       unlink(tmpfile.c_str());
+     } catch (ReadError) {
+       cout << "\"" << cmd << "\" failed - skipping\n";
+       unlink(tmpfile.c_str());
+       return;
+     } catch (...) {
+       unlink(tmpfile.c_str());
+     }
+
+   The ``shell_protect`` function escapes shell meta-characters in the filename. The ``stdout_to_string`` function runs a command and captures its output as a C++ std::string. If the command is not installed on PATH, omindex detects this automatically and disables support for the mimetype in the current run, so it will only try the first file of each such type.
+
+   If UTF-8 output is not supported, pick the best (or only!) supported encoding and then convert the output to UTF-8 - to do this, once you have dump, convert it like so (replacing "UTF-16" with the character set which is produced)
+   ::
+
+     convert_to_utf8(string, "UTF-16");
+
+If you find a reliable external filter or library and think it might be useful to other people, please let us know about it.
+
+Submitting a patch:
+===================
+
+Once you are happy with how your handler/filter works, please submit a patch so we can include it in future releases (creating a new trac ticket and attaching the patch is best). Before doing so, please also update docs/overview.rst by:
+
+- Adding the format and extensions recognised for it to the list.
+- Adding the mime-type to 'mimemap.tokens'.
+
+It would be really useful if you are able to supply some sample files with a licence which allows redistribution so we can test the filters on it. Ideally ones with non-ASCII characters so that we know Unicode support works.
+
+You can read more about how to contribute to Xapian `here <https://xapian-developer-guide.readthedocs.io/en/latest/contributing/index.html>`_.
```

Comment:
Try to avoid link text like "here" or "this" links - better to write "You can read more about how to contribute to Xapian" and put the link on "how to contribute to Xapian".

E.g. see: https://www.w3.org/TR/WCAG10-HTML-TECHS/#link-text

Reply:
I am going to try to avoid this kind of link text.